### ISSUE TITLE
Error: Could not find or load main class org.acoli.conll.rdf.

### ISSUE THREAD

**Issue by AlanQuille** [acoli-repo/conll-rdf#64]
Hi everyone,
I'm trying to execute the command "./run.sh" but I get the error shown in the title. I tried to run ./compile.sh, initially it was unsuccessful because quotation marks around the variable $1 were missing on line 18 of compile.sh. I fixed this issue and compile.sh runs successfully. But I still get the error shown in the title.

I am using git bash on Windows 10. I am trying to do this in an anaconda environment with openjdk 8 installed, Maven version 3.6.3.

**Comment by leogott** [comment]
Hey @AlanQuille and welcome to the conll-rdf repo!

Your problem is likely caused by not specifying which class (tool) to run. Please have a look at the "Getting Started" Section in our Readme and check if that's the case.

**Comment by AlanQuille** [comment]
> 
> 
> Hey @AlanQuille and welcome to the conll-rdf repo!
> 
> Your problem is likely caused by not specifying which class (tool) to run. Please have a look at the "Getting Started" Section in our Readme and check if that's the case.

Hi, I think I specified the class path in the following command:

cat irishtimes.conllu | ./run.sh CoNLLStreamExtractor https://github.com/UniversalDependencies/UD_English# ID WORD LEMMA UPOS POS FEAT HEAD EDGE DEPS MISC | ./run.sh CoNLLRDFFormatter -rdf > irishtimes.ttl

... but I got the following error:

Error: Could not find or load main class org.acoli.conll.rdf.CoNLLStreamExtractor
Error: Could not find or load main class org.acoli.conll.rdf.CoNLLRDFFormatter
(java8)

**Comment by AlanQuille** [comment]
Just to clarify, I am running this command in the conll-rdf folder, not in any of the subdirectories.

**Comment by AlanQuille** [comment]
Also tried this command, got this error:
cat irishtimes.conllu | ./run.sh CoNLLRDFFormatter > irishtimes.ttl

**Comment by leogott** [comment]
Thank you for specifying. You are using the run-script correctly.
There seems to be an issue with the compiling of classes.

You said you were running conll-rdf via git-bash on windows 10.

Could you please tell me the results of running the commands `javac -version` and `mvn -version`.

And if at least javac is installed, you could try running `./compile -f` and see if that changes anything about the problem.

That should give me enough information to either help you with your problem or reproduce it locally.

**Comment by AlanQuille** [comment]
Thank you for replying. I tried ./compile.sh -f and unfortunately it didn't change anything.

I ran the following commands, the results are output here:

$javac -version
javac 1.8.0_152-release
(java8)

$ mvn -version
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: C:\Users\Alan\.conda\envs\java8\Library
Java version: 15.0.1, vendor: Oracle Corporation, runtime: C:\Program Files\Java\jdk-15.0.1
Default locale: en_IE, platform encoding: UTF8
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
(java8)

**Comment by leogott** [comment]
Okay, so the problem is an interesting one.
tldr: compile.sh uses maven, which compiles for java 15, but run.sh uses plain `java`, which is java 8.
This is relatively easy to fix, by making run.sh default to maven as well (`mvn exec:java -Dexec.mainClass="org.acoli.conll.rdf.CoNLLStreamExtractor" -Dexec.args="https://github.com/UniversalDependencies/UD_English# ID WORD LEMMA UPOS POS FEAT HEAD EDGE DEPS MISC"`).

However, maven should use java 8 to compile conll-rdf, because the compile target is specified in the pom of the project.
So this issue would be caused by maven not knowing about your java 8 installation, I think?

**Comment by AlanQuille** [comment]
I looked online and I saw it's common for mvn to find the incorrect version of Java. I will fix this and update with results, thanks again for your help.

**Comment by leogott** [comment]
You're welcome!

https://github.com/acoli-repo/conll-rdf/blob/34e036f1bccaa479ce8e65524f35b0e7ae980c9f/run.sh#L16
you can also work around your issue by replacing this line with
`mvn exec:java -Dexec.mainClass="org.acoli.conll.rdf.$1" -Dexec.args="${*:2}" | \`
(barely tested but appears to do what it should)

**Comment by AlanQuille** [comment]
Hi @leogott, I have fixed the issue, java -version and mvn -version are now showing identical java versions. I also did the replacement in run.sh as you suggested. However, I am now faced with this error:

![image](https://user-images.githubusercontent.com/33432976/128201349-479c7b07-29ad-4f2f-af66-c6c247e2b2c7.png)

**Comment by AlanQuille** [comment]
I added the -e tag as suggested to get a full stack trace and this was the result:

![image](https://user-images.githubusercontent.com/33432976/128201950-64d6f86c-5cb7-4be1-a860-aa0dec786a80.png)

**Comment by leogott** [comment]
The error message indicates that no main class was passed to maven (via run.sh) at all. Please also tell me the command you entered.

To eliminate a common source of problems, please delete your conll-rdf/target directory before trying again.

If that doesn't change anything, please revert your run.sh to the original state and try again.

**Comment by AlanQuille** [comment]
Hi @leogott, I got it working with this command: 

cat irishtimes.conllu | ./run.sh CoNLLStreamExtractor https://github.com/UniversalDependencies/UD_English# ID WORD LEMMA UPOS POS FEAT HEAD EDGE DEPS MISC | ./run.sh CoNLLRDFFormatter -rdf > irishtimes.ttl


I have one final question: what is the command to convert the irishtimes.ttl to Conll-2012 format? The command shown [https://github.com/acoli-repo/conll-rdf/blob/master/examples/README.md](url) converts the .ttl file back to Conll-U format but that is not what I need. Thanks again.

**Comment by AlanQuille** [comment]
Hi @leogott, I used this command and it is now working, thank you very much:

cat irishtimes.conllu | ./run.sh CoNLLStreamExtractor https://github.com/UniversalDependencies/UD_English# ID WORD LEMMA UPOS POS FEAT HEAD EDGE DEPS MISC | ./run.sh CoNLLRDFFormatter -rdf > irishtimes.ttl
